datumaro/plugins/icdar_format/converter.py

```diff
@@ -0,0 +1,193 @@
+# Copyright (C) 2020 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+import os
+import os.path as osp
+
+from datumaro.components.converter import Converter
+from datumaro.components.extractor import AnnotationType, CompiledMask
+from datumaro.util.annotation_util import find_instances
+from datumaro.util.image import save_image
+from datumaro.util.mask_tools import find_mask_bbox, paint_mask
+
+from .format import IcdarPath, IcdarTask
+
+
+class _WordRecognitionConverter():
+    def __init__(self):
+        self.annotations = ''
+
+    def save_annotations(self, item):
+        self.annotations += '%s, ' % (item.id + IcdarPath.IMAGE_EXT)
+        for ann in item.annotations:
+            if ann.type != AnnotationType.caption:
+                continue
+            self.annotations += '\"%s\"' % ann.caption
+        self.annotations += '\n'
+
+    def write(self, path):
+        file = osp.join(path, 'gt.txt')
+        os.makedirs(osp.dirname(file), exist_ok=True)
+        with open(file, 'w') as f:
+            f.write(self.annotations)
+
+    def is_empty(self):
+        return len(self.annotations) == 0
+
+class _TextLocalizationConverter():
+    def __init__(self):
+        self.annotations = {}
+
+    def save_annotations(self, item):
+        annotation = ''
+        for ann in item.annotations:
+            if ann.type == AnnotationType.bbox:
+                annotation += ' '.join(str(p) for p in ann.points)
+                if ann.attributes and 'text' in ann.attributes:
+                    annotation += ' \"%s\"' % ann.attributes['text']
+            elif ann.type == AnnotationType.polygon:
+                annotation += ','.join(str(p) for p in ann.points)
+                if ann.attributes and 'text' in ann.attributes:
+                    annotation += ',\"%s\"' % ann.attributes['text']
+            annotation += '\n'
+        self.annotations[item.id] = annotation
+
+    def write(self, path):
+        os.makedirs(path, exist_ok=True)
+        for item in self.annotations:
+            file = osp.join(path, 'gt_' + item + '.txt')
+            with open(file, 'w') as f:
+                f.write(self.annotations[item])
+
+    def is_empty(self):
+        return len(self.annotations) == 0
+
+class _TextSegmentationConverter():
+    def __init__(self):
+        self.annotations = {}
+        self.masks = {}
+
+    def save_annotations(self, item):
+        annotation = ''
+        colormap = [(255, 255, 255)]
+        masks = [a for a in item.annotations
+            if a.type == AnnotationType.mask]
+        annotations = {}
+        for ann in find_instances(masks):
+            if ann[0].group in annotations:
```

Comment:
Group=0 means 'no group', so such annotations should not be joined.

Reply:
Probably, I need to correct the group numbering in the tests and in the extractor. But here items with `group = 0` will just be a separate group.

Reply:
It's just a group of one item.

Reply:
In our case, the group also shows the order in the text annotation. If we make it so that each one item group = 0, when converting, we get a different order in the annotation. I think this is not very good.